koku/masu/external/downloader/aws/aws_report_downloader.py

```diff
@@ -103,30 +111,35 @@ def create_daily_archives(
     s3_csv_path = com_utils.get_path_prefix(
         account, Provider.PROVIDER_AWS, provider_uuid, start_date, Config.CSV_DATA_TYPE
     )
-    data_frame, time_interval, process_date = get_initial_dataframe_with_date(
+    use_cols, time_interval, process_date = get_initial_dataframe_with_date(
         local_file, s3_csv_path, manifest_id, provider_uuid, start_date, end_date, context, tracing_id
     )
-    intervals = data_frame[time_interval].unique()
-    for interval in intervals:
-        date = interval.split("T")[0]
-        csv_date = datetime.datetime.strptime(date, "%Y-%m-%d")
-        # Adding end here so we dont bother to process future incomplete days (saving plan data)
-        if csv_date >= process_date and csv_date <= end_date:
-            dates.add(date)
-    if not dates:
-        return [], {}
-    directory = os.path.dirname(local_file)
-    date_range = {"start": min(dates), "end": max(dates), "invoice_month": None}
-    data_frame = data_frame[data_frame[time_interval].str.contains("|".join(dates))]
-    for date in dates:
-        daily_data = data_frame[data_frame[time_interval].str.match(date)]
-        day_file = ReportManifestDBAccessor().update_and_get_day_file(date, manifest_id)
-        day_filepath = f"{directory}/{day_file}"
-        daily_data.to_csv(day_filepath, index=False, header=True)
-        utils.copy_local_report_file_to_s3_bucket(
-            tracing_id, s3_csv_path, day_filepath, day_file, manifest_id, start_date, context
-        )
-        daily_file_names.append(day_filepath)
+    LOG.info(log_json(tracing_id, msg="pandas read csv with following usecols", usecols=use_cols, context=context))
+    with pd.read_csv(local_file, usecols=use_cols, chunksize=settings.PARQUET_PROCESSING_BATCH_SIZE) as reader:
```

Comment:
So, I have an idea that I wondering if it would be okay with it if I ran with it after this gets merged?

Part of me is wondering if we just want to steal the an entire with statement we use [here](https://github.com/project-koku/koku/blob/main/koku/masu/processor/parquet/parquet_report_processor.py#L522-L524). That way we can do the csv converters ahead of time too. I wonder if this would fix some of the weird date stuff you have to do later on. 

I think there are more elements in the PostProcessor that we could pull out and create a PreProcessor class?

Reply:
Yeh I was actually thinking something similar too with the converters. Certainly seems silly opening things thn opening them again to sort the converters.